src/server/graphql/graphql.js

```diff
@@ -0,0 +1,58 @@
+import {parse} from 'graphql/language/parser'
+import {execute} from 'graphql/execution/execute'
+import {validateSchema} from 'graphql/type/validate'
+import {validate} from 'graphql/validation/validate'
+
+// Avoid needless parsing & validating for the 300 hottest operations
+let documentCache = {}
+const MAX_CACHE_SIZE = 300
+
+const graphql = async (
```

Comment:
since the refactor, this isn't necessary for rate limiting, but it's a big performance win.

For example, on every update drag mutation, graphql had to create a new AST & validate it. now it can reuse the same AST & skip validation.